### PR TITLE
Refactoring Documentations.

### PR DESCRIPTION
**Pull request by nhosoi**
- Eliminating redundant contents from top README.md and roles/rsyslog/README.md.
- Adding the description for CI-tests and manual tests.
- Adding the way how to add new input/output to the logging role.
- Keeping old top README.md and roles/rsyslog/README.md as OLDREADME.md, which
  will be eventually removed.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,408 +1,41 @@
...
-========================================
+# linux-system-roles/logging
 
-Guidelines for Using Logging Ansible Role
------------------------------------------
+## Background
 
-The `logging` role enables a RHEL admin/developer to deploy logging collectors on the local host, remote host or set of remote hosts,
-process these logs if needed to add additional metadata and ship it to a remote location to be saved and analyzed.
+Logging Role is an abstract layer for provisioning and configuring the logging system. Currently, rsyslog is only a supported provider.
```
Comment (richm) [inline README.md:5]: "Currently, rsyslog is the only supported provider."

File: README.md [hunk 2]
```diff
@@ -1,408 +1,41 @@
...
-===========
-
-  - [`Rsyslog`](https://www.rsyslog.com/) - The logging role default log collector used for log processing.
-  - [`Viaq`](https://docs.okd.io/latest/install_config/aggregate_logging.html)- Common Logging based on OpenShift Aggregated Logging (OCP/Origin).
-  - [`Elasticsearch`](https://www.elastic.co/) - Non-OpenShift standalone Elasticsearch.
-  - `Local` - Output the collected logs to a local File / Journal (Not yet implemented). Supported only for default and basics Rsyslog data at this point.
-  - `Remote Rsyslog` - Input logs from or output logs to a remote Rsyslog server.
+  - `logging_inputs` - List of logging inputs dictionary to specify input types.
+    * `basics` - basic inputs configuring inputs from systemd journal or unix socket.
+    * `fiels` - files inputs configuring inputs from local files.
```
Comment (richm) [inline README.md:15]: files

File: README.md [hunk 3]
```diff
@@ -1,408 +1,41 @@
...
-      * [playbook.yml](#playbookyml)
-   * [New Projects Integration](#new-projects-integration)
-      * [Planned Flows](#planned-flows)
-   * [Testing](#testing)
-   * [Additional Resources](#additional-resources)
-   * [License](#license)
-<!--te-->
+## Definitions
 
-Definitions
```
Comment (richm) [inline README.md:32]: You need to have a section that specifies the variables used as input parameters to the logging role - see https://github.com/linux-system-roles/firewall#variables - you can also have a `Definitions` section where you define the terms used.  You should also have a section https://github.com/linux-system-roles/firewall#example-playbooks with some short examples.

File: roles/rsyslog/README.md [hunk 4]
```diff
@@ -1,408 +1,434 @@
...
+    outputs: [output_name0, output_name1]
+  - name: flow_nameY
+    inputs: [input_nameB]
+    outputs: [output_name1]
 ```
 
-vars.yml
----------
-vars.yml stores variables which are passed to ansible to control the tasks.  The contents of this file could be merged into the inventory file.
+## Logging_inputs parameters
```
Comment (richm) [inline roles/rsyslog/README.md:49]: This should be in the `Variables` section in the `logging/README.md`

File: roles/rsyslog/README.md [hunk 5]
```diff
@@ -1,408 +1,434 @@
...
+  - `outputs`: A list of outputs. to which the log messages are sent.
+
+## Other parameters
+
+These parameters are set in the same level of the `logging_inputs`, `logging_output`, and `logging_flows`.
+
+- `logging_enabled`: When 'true', logging role will deploy specified configuration file set. Default to 'true'.
+- `logging_pki`: Specifying an encryption. One of `none`, `ptcp`, `tls`, `gtls`, and `gnutls`. Default to `ptcp`.
+                 Note: `none`=`ptcp`, `tls`=`gtls`=`gnutls`.
+                 When logging_pki is _not_ `ptcp`, rsyslog_pki_path`, `rsyslog_pki_realm`, `rsyslog_pki_ca`, `rsyslog_pki_crt`, `rsyslog_pki_key` are configured.
```
Comment (richm) [inline roles/rsyslog/README.md:177]: missing ` in front of rsyslog_pki_path

File: README.md (hunk 3 above)
Comment (nhosoi) [reply README.md:32]: > You need to have a section that specifies the variables used as input parameters to the logging role - see https://github.com/linux-system-roles/firewall#variables - you can also have a `Definitions` section where you define the terms used. You should also have a section https://github.com/linux-system-roles/firewall#example-playbooks with some short examples.

Somehow I put some of them (maybe not all though) in the rsyslog README. Do you think I should move them to the top README?

Comment (richm) [reply README.md:32]: > Do you think I should move them to the top README?

Yes.  All public parameters should be in the top level `logging/README.md`, and should be specified in the top level `logging/defaults/main.yml`.

**Comment by nhosoi** [comment]
Sorry, if the 3rd commit https://github.com/linux-system-roles/logging/pull/132/commits/f2dd1cb3bd685b4f4065963f12379867e3ff9ec0 had better be in a separate pr, I'm opening another. (The 3rd commit depends on the 1st commit. So, if I could merge the 1st and 2nd commits now, that'd help me to open a new pr from the 3rd commit. Thanks! ;)

Note: The 3rd commit https://github.com/linux-system-roles/logging/pull/132/commits/f2dd1cb3bd685b4f4065963f12379867e3ff9ec0 is localized in the logging/tests directory and I'd like to include this "Cleaning up CI tests" in the release asap to share them with Dalibor.

**Comment by nhosoi** [comment]
[citest bad]

File: tests/tests_basics_files2.yml [hunk 6]
```diff
@@ -53,42 +74,33 @@
...
-      register: files_conf_messages
+    - name: Check the config file contains the expected filter and the output file as configured.
+      command: /bin/grep '\*.info;authpriv.none;auth.none;cron.none;mail.none.*{{ __default_system_log }}' '{{ __test_files_conf }}'
+      register: __result
       changed_when: false
-
-    - name: Check the filter
-      assert:
-        that: files_conf_messages.stdout is match(" *\*.info;*authpriv.none;auth.none;cron.none;mail.none.*.var.log.messages")
+      failed_when: __result.rc != 0
```
Comment (richm) [inline tests/tests_basics_files2.yml:96]: Use `failed_when: __result is failed` - see https://docs.ansible.com/ansible/latest/user_guide/playbooks_tests.html#task-results

But for this case, do you need a `register` and a `failed_when`?  That is, if `command` returns with an error, the task will fail, and you don't use `__result` elsewhere.

File: tests/tests_basics_files_forwards.yml [hunk 7]
```diff
@@ -64,54 +88,39 @@
     # but some functionality is disabled due to some error, e.g., error: 'tls.cacert' file couldn't be accessed.
     - name: Check rsyslog errors
       command: systemctl status rsyslog
-      register: systemctl_result
-      failed_when: "'error' in systemctl_result.stdout or 'a RainerScript command' in systemctl_result.stdout or systemctl_result.rc != 0"
+      register: __result
+      failed_when: "'error' in __result.stdout or 'a RainerScript command' in __result.stdout or __result.rc != 0"
```
Comment (richm) [inline tests/tests_basics_files_forwards.yml:92]: use `__result is failed` instead of `__result.rc != 0`

**Comment by nhosoi** [comment]
Thank you, @richm. Squashed into 2 commits. Merging.

**Pull request merged**